### ISSUE TITLE
Reproduction results are relatively poor

### ISSUE THREAD

**Issue by Miraclerice** [Junelin2333/LanGuideMedSeg-MICCAI2023#3]
Hello, I would like to ask you a question. I reproduced the code according to your repo. The dice of the experimental results I ran out are about 4 different from the results of your paper. Is there anything I need to pay attention to?
 @Junelin2333

**Comment by Junelin2333** [comment]
I noticed what you said about this. Please check if the results obtained on the test set. If valid set, the result is about 0.86 (Dice Score)

**Comment by Miraclerice** [comment]
Since running the training script reported an error on line 55 of the training script saying that the task parameter was missing, I added that parameter to Accuracy, not sure if that is correct. Here are my results after training, it doesn't show up very high. Could it be because my dataset is set up incorrectly?
![image](https://github.com/Junelin2333/LanGuideMedSeg-MICCAI2023/assets/100824907/ff2f8060-d0e4-43c6-b930-1a12099c4791)
![image](https://github.com/Junelin2333/LanGuideMedSeg-MICCAI2023/assets/100824907/6f6c3e1e-111c-4cb1-b6f8-b6c8efc36c92)
![image](https://github.com/Junelin2333/LanGuideMedSeg-MICCAI2023/assets/100824907/9ecb611d-a74e-425a-a9ef-b07af882ccfc)

**Comment by Miraclerice** [comment]
If I'm doing it wrong, I'd appreciate guidance and a solution! @Junelin2333

**Comment by Junelin2333** [comment]
About error on line 55, it's due to the version of Torchmetrics, use parameters `task='binary'` is also work, I have update the code in `train.py` and move some code to `engine/wrapper.py`.

I'm not sure if you want to ask about the discrepancy between the results in Table 1 and the _epoch 53 result_ in your picture.
If yes, I have noted that the result should be obtained on the test set rather than valid set. 

I have upload a file named `evaluate.py`, please referring the file and the **readme** to get the result on the test set.

Best wishes :)

**Comment by Miraclerice** [comment]
Thank you very much for your answers and repo updates, I hope you are getting better and better in your research journey! @Junelin2333

**Comment by Junelin2333** [comment]
Don't forget give me a star if my anwsers have helped you! Thanks! Best wishes to your academic, too!
